constraint/pkg/externaldata/request.go

```diff
@@ -100,44 +100,43 @@ func DefaultSendRequestToProvider(ctx context.Context, provider *v1alpha1.Provid
 	return &externaldataResponse, resp.StatusCode, nil
 }
 
-// getClient returns a new HTTP client, and set up its TLS configuration if necessary.
-func getClient(provider *v1alpha1.Provider, clientCert *tls.Certificate) (*http.Client, error) {
+// getClient returns a new HTTP client, and set up its TLS configuration.
+func getClient(provider *unversioned.Provider, clientCert *tls.Certificate) (*http.Client, error) {
 	u, err := url.Parse(provider.Spec.URL)
 	if err != nil {
 		return nil, fmt.Errorf("failed to parse provider URL %s: %w", provider.Spec.URL, err)
 	}
 
+	if u.Scheme != HTTPSScheme {
+		return nil, fmt.Errorf("only HTTPS scheme is supported")
+	}
+
 	client := &http.Client{
 		Timeout: time.Duration(provider.Spec.Timeout) * time.Second,
 	}
 
-	tlsConfig := &tls.Config{
-		//nolint:gosec
-		InsecureSkipVerify: provider.Spec.InsecureTLSSkipVerify,
-	}
+	tlsConfig := &tls.Config{MinVersion: tls.VersionTLS13}
```

Comment:
Should min TLS version be configurable? we should at least add this to the docs.

Reply:
I think if we are making this secure by default. It's okay to set min as 1.3 to match GK. I'll add to docs.

Reply:
For new integrations, there is no reason to not use TLS 1.3.  TLS 1.2 requires careful cipher control to get a secure config.  It is [a lot of work](https://github.com/vmware-tanzu/pinniped/blob/main/internal/crypto/ptls/default.go).

Reply:
added to docs: [open-policy-agent/gatekeeper@`767cd9e` (#2438)](https://github.com/open-policy-agent/gatekeeper/pull/2438/commits/767cd9e9ca952160e0f54dccb4562694efabe7fc)